build/updateCitation.js

```diff
@@ -0,0 +1,25 @@
+import { fileURLToPath }            from 'url'
+import yaml                         from 'js-yaml'
+import { readFile, writeFile }      from 'fs/promises'
+
+import {
+  dirname as getDirname,
+  join as joinPath,
+} from 'path'
+
+const __filename  = fileURLToPath(import.meta.url)
+const __dirname   = getDirname(__filename)
+
+const citationPath = joinPath(__dirname, `../CITATION.cff`)
+const citationText = await readFile(citationPath, `utf8`)
+
+const packagePath = joinPath(__dirname, `../package.json`)
+const packageText = await readFile(packagePath, `utf8`)
+
+const citationObject = yaml.load(citationText)
+const packageObject =  yaml.load(packageText)
+citationObject['date-released'] = new Date().toLocaleDateString('en-CA')
+citationObject['version'] = packageObject['version']
```

Comment:
The reason we used [bracket notation] for the `date-released` variable is because it has a hyphen in it. Bracket notation is the only way to access an object property that has special characters in it. But the `version` property is just a normal property, so you can access it with `citationObject.version`.

```suggestion
citationObject['date-released'] = new Date().toLocaleDateString('en-CA')
citationObject.version = packageObject.version
```